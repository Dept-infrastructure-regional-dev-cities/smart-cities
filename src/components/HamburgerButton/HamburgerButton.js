import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import style from './HamburgerButton.scss';

const HamburgerButton = ({onClick, _className, label}) => {
  const className = classNames(
    style.hamburgerButton,
    _className,
  );

  return (
    <button
      className={className}
      onClick={onClick}
      aria-label={label}
    >
      <div className={style.hamburgerButton__hamburger_icon}>
        <div className={style.hamburgerButton__hamburger_icon_bar} />
        <div className={style.hamburgerButton__hamburger_icon_bar} />
        <div className={style.hamburgerButton__hamburger_icon_bar} />
      </div>
    </button>
  )
};

HamburgerButton.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

HamburgerButton.defaultProps = {
  label: 'open menu',
};

export default HamburgerButton;
