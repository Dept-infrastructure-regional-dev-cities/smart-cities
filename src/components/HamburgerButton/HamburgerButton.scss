@import '~style/variables';

.wrapper {
  width: $app_header_height_mobile;
  height: $app_header_height_mobile;
  background: $color-primary--600;
  border: none;
  border-radius: 0;
}

.icon {
  display: flex;
  flex-direction: column;
}

.bar {
  background: white;
  height: 4px;
  border-radius: 2px;
  margin: 3px;
}
