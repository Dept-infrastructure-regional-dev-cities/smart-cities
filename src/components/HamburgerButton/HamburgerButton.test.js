import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HamburgerButton from './HamburgerButton';

it('should render consistently', () => {
  const tree = renderer.create(
    <HamburgerButton onClick={() => {}}/>
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should call onClick when clicked', () => {
  const onClickMock = jest.fn();

  const tree = shallow(
    <HamburgerButton onClick={onClickMock}/>
  );

  tree.find('button').simulate('click');

  expect(onClickMock).toHaveBeenCalledTimes(1);
});
