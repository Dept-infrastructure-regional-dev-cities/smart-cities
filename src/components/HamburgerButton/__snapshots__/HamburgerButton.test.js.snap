// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render consistently 1`] = `
<button
  aria-label="open menu"
  className="hamburgerButton"
  onClick={[Function]}
>
  <div
    className="hamburgerButton__hamburger_icon"
  >
    <div
      className="hamburgerButton__hamburger_icon_bar"
    />
    <div
      className="hamburgerButton__hamburger_icon_bar"
    />
    <div
      className="hamburgerButton__hamburger_icon_bar"
    />
  </div>
</button>
`;
